crates/wasi/src/clocks.rs

```diff
@@ -141,3 +155,56 @@ impl TryFrom<SystemTime> for Datetime {
         })
     }
 }
+
+pub struct Timezone {
+    // The underlying system timezone.
+    timezone: cap_time_ext::Timezone,
+}
+
+impl Default for Timezone {
+    fn default() -> Self {
+        Self::new(ambient_authority())
+    }
+}
+
+impl Timezone {
+    pub fn new(ambient_authority: AmbientAuthority) -> Self {
+        Self {
+            timezone: cap_time_ext::Timezone::new(ambient_authority),
+        }
+    }
```

Comment:
I feel like this interface would have been useful if embedders were able to provide an arbitrary timezone to use with UTC default.

There are many ways to do that, but perhaps we could:
- store `jiff::tz::TimeZone` https://docs.rs/jiff/latest/jiff/tz/struct.TimeZone.html in the struct
- provide `TryFrom<cap_time_ext::Timezone> for Timezone`
- provide `From<jiff::tz::Timezone> for Timezone`

If we had that, we could have a `Default` implemented for `Timezone`, which would default to UTC.

We could then:
- rename `new` to something like `new_host` and make it fallible
- add `new_utc`, which acts the same as `Default`
- maybe implement `FromStr` for `Timezone` in terms of https://docs.rs/jiff/latest/jiff/tz/struct.TimeZone.html#method.get ?

Reply:
For me it's not necessarily a requirement to change cap-std bits, it's fine to use jiff directly here in Wasmtime

Reply:
I'd prefer to wasmtime-wasi jiff directly instead of going through cap-std-ext, I don't think the indirection through that additional crate is really getting us much in this case. The user can provide.a jiff::Timezone to the WasiCtxBuilder and if they don't it should default to UTC. We can then have a wasmtime-cli flag for setting a time zone, or "auto" which allows jiff to make its best effort to figure one out.

Reply:
I'd expect to see something like:

```suggestion
pub struct Timezone {
    // The underlying system timezone.
    timezone: jiff::tz::Timezone,
}

impl Default for Timezone {
    fn default() -> Self {
        Self::from(jiff::tz::Timezone::UTC)
    }
}

impl From<jiff::tz::Timezone> for Timezone {
   fn from(timezone: jiff::tz::Timezone) -> Self {
       Self {
          timezone,
       }
   }
}
```

Personally, I am wary of exposing third-party library dependencies in public API, especially pre-1.0, since that could cause hard to debug issues downstream if/when `jiff` version does not match for users. That's why I'm proposing `FromStr` implemntation via `get`, such that users can do `let tz: Timezone = "Europe/Amsterdam"::parse()?`

Reply:
Thanks. That answers my question. We can get rid of the ambient authority.

Reply:
> I'd prefer to wasmtime-wasi jiff directly instead of going through cap-std-ext, I don't think the indirection through that additional crate is really getting us much in this case. The user can provide.a jiff::Timezone to the WasiCtxBuilder and if they don't it should default to UTC. We can then have a wasmtime-cli flag for setting a time zone, or "auto" which allows jiff to make its best effort to figure one out.

Yes. Jiff seems to be the way to go. I've updated the PR to reflect that. 

I think allowing the user to override the system timezone is reasonable, especially for testing purposes, but I don't see why we would want to default to UTC if that timezone is not provided. Reporting the correct timezone would be the expected behavior for most applications and would be consistent with native code. For edge compute applications, I would imagine the automatic reporting of the correct timezone is a table stakes expectation.

Reply:
If we want to default to UTC, then I don't see any role for wasmtime at all. There is no host functionality that can be provided.

Reply:
I fear you might be misunderstanding me, but to clarify I'd think of this like environment variables. By default wasmtime doesn't give any environment variables to the guest, but the `WasiCtxBuilder` structure has methods to pass environment variables. The default behavior of "nothing" doesn't mean that the implementation of environment variabls is pointless it's just the default behavior.

I'm thinking the same thing for time zones. By default it's UTC, but embedders can easily configure "use this time zone instead" where "this" could be something specific or the system's local time zone. I'm not sure why this behavior would lead to the implementation having no role in Wasmtime. Wasmtime is still providing functionality, time zones and information, and embedders can configure which time zone is reported.

Reply:
This would probably be easily smoothed over in a quick chat, but I'll try to articulate my understanding as best I can here. Today, there is nothing technically stopping someone from using timezones in their component and setting the timezone to UTC or whatever timezone they want. If they want to enable host timezone access, they would need something like what this PR provides.
If we want that functionality not to be enabled by default then that's fine. Perhaps that is what you mean. 

Where I'm coming from people expect their Wasm applications to work just like their native code. For C2PA, all of our tests run with `-S cli -S http --dir .`. Too many WASI specific code paths and it just isn't worth the hassle to maintain.
If `--dir` isn't passed and and file access is denied, then the code panics. Likewise if people don't want the system timezone, then they shouldn't pass the feature flag. If they do want it, then it should just work as like the native code, which is their expectation.

Reply:
Could you help me understand why we'd need different code paths?

For the CLI you'd pass something like `--inherit-timezone` to use the timezone of the host or `--timezone $mytz` to use some custom one. If you don't specify any - it runs in UTC by default.

In case of an embedding you'd use analogous options on the `WasiCtxBuilder`

If this functionality does not address your use case, I'd really like to know more about it